torch/_lowrank.py

```diff
@@ -57,27 +58,27 @@ def get_approximate_basis(
     """
 
     niter = 2 if niter is None else niter
-    m, n = A.shape[-2:]
-    dtype = _utils.get_floating_dtype(A)
+    dtype = _utils.get_floating_dtype(A) if not A.is_complex() else A.dtype
     matmul = _utils.matmul
 
-    R = torch.randn(n, q, dtype=dtype, device=A.device)
+    R = torch.randn(A.shape[-1], q, dtype=dtype, device=A.device)
 
     # The following code could be made faster using torch.geqrf + torch.ormqr
     # but geqrf is not differentiable
-    A_H = _utils.transjugate(A)
-    if M is None:
-        Q = torch.linalg.qr(matmul(A, R)).Q
-        for i in range(niter):
-            Q = torch.linalg.qr(matmul(A_H, Q)).Q
-            Q = torch.linalg.qr(matmul(A, Q)).Q
-    else:
-        M_H = _utils.transjugate(M)
-        Q = torch.linalg.qr(matmul(A, R) - matmul(M, R)).Q
-        for i in range(niter):
-            Q = torch.linalg.qr(matmul(A_H, Q) - matmul(M_H, Q)).Q
-            Q = torch.linalg.qr(matmul(A, Q) - matmul(M, Q)).Q
 
+    X = matmul(A, R)
+    if M is not None:
+        X = X - matmul(M, R)
+    Q = torch.linalg.qr(X).Q
+    for i in range(niter):
+        X = matmul(A.mH, Q)
```

Comment:
No need to recompute `A.mH` each iteration? (I know it's just a view but still)

Reply:
I mean, it's a view in a linalg operation that does lots of matmuls, a svd, etc. I prefer readability here.

Reply:
Each tensor takes about 1 us to create, and for small sizes `svd_lowrank` takes on the order of 100s of us so we're talking about a 1% speedup for 1 line of code. I would take that trade personally, but maybe that's just me.

Reply:
Having
```python
A_H = A.mH
for i in range(niter):
    X = matmul(A_H, Q)
```
does not sacrifice readability, IMHO.

Reply:
This function is to be run on very large matrices that `linalg.svd` cannot support due to OOM

Reply:
How this comments about OOM relates to moving a view operation out of the loop? AFAIK, there is no additional burden to memory allocations.

Reply:
My point is that Peter's comment is moot. This function is to be called with huge matrices, so all this discussion is just about a premature optimisation.

Reply:
Got it!